controlplane/pkg/model/dataplane.go

```diff
@@ -62,24 +64,29 @@ func newDataplaneDomain() dataplaneDomain {
 	}
 }
 
-func (d *dataplaneDomain) AddDataplane(dp *Dataplane) {
-	d.store(dp.RegisteredName, dp)
+func (d *dataplaneDomain) AddDataplane(dp *Dataplane) error {
+	if ok := d.store(dp.RegisteredName, dp, false); ok {
+		return nil
+	}
+	return fmt.Errorf("trying to add dataplane by existing name: %v", dp.RegisteredName)
+}
+
+func (d *dataplaneDomain) AddOrUpdateDataplane(dp *Dataplane) {
+	d.store(dp.RegisteredName, dp, true)
 }
 
 func (d *dataplaneDomain) GetDataplane(name string) *Dataplane {
-	v, _ := d.load(name)
-	if v != nil {
+	if v, ok := d.load(name); ok {
 		return v.(*Dataplane)
 	}
 	return nil
 }
 
-func (d *dataplaneDomain) DeleteDataplane(name string) {
-	d.delete(name)
-}
-
-func (d *dataplaneDomain) UpdateDataplane(dp *Dataplane) {
-	d.store(dp.RegisteredName, dp)
+func (d *dataplaneDomain) DeleteDataplane(name string) error {
+	if ok := d.delete(name); ok {
```

Comment:
[minor] looks like delete returns boolean. Can we remove ok variable?

Reply:
Done : a7226f2